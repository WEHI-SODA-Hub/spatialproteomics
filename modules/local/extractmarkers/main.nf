process EXTRACTMARKERS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container 'community.wave.seqera.io/library/tifffile_pandas_typer:8947a22261c53cf2'

    input:
    tuple val(meta), path(tiff)

    output:
    tuple val(meta), path("${prefix}_markers.csv"), emit: markers
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    extract_markers.py \\
        $args \\
        $tiff > ${prefix}_markers.csv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_markers.csv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //')
    END_VERSIONS
    """
}
