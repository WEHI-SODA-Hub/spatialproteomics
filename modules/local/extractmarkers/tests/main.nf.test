nextflow_process {

    name "Test Process EXTRACTMARKERS"
    script "../main.nf"
    process "EXTRACTMARKERS"

    tag "modules"
    tag "modules_local"
    tag "extractmarkers"

    test("extractmarkers - tiff") {

        when {
            process {
                // NOTE: This test file doesn't actually work, but I don't have a test comet image at the moment
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file('https://github.com/nf-core/test-datasets/raw/modules/data/imaging/segmentation/nuclear_image.tif', checkIfExists: true)

                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("extractmarkers - tiff - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file('https://github.com/nf-core/test-datasets/raw/modules/data/imaging/segmentation/nuclear_image.tif', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
