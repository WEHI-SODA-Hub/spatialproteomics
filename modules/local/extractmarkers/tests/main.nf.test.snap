{
    "extractmarkers - tiff - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": [
                                "analyse_samplesheet.csv:md5,bf995bff1feac8b4fc2b4f46e7ae5b08"
                            ]
                        },
                        "test_markers.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e1cb9bf9ff95b194727697ba4789b26e"
                ],
                "markers": [
                    [
                        {
                            "id": [
                                "analyse_samplesheet.csv:md5,bf995bff1feac8b4fc2b4f46e7ae5b08"
                            ]
                        },
                        "test_markers.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e1cb9bf9ff95b194727697ba4789b26e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-19T12:12:23.296865602"
    },
    "extractmarkers - tiff": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": [
                                "analyse_samplesheet.csv:md5,bf995bff1feac8b4fc2b4f46e7ae5b08"
                            ]
                        },
                        "test_markers.csv:md5,4b612101a8ba4a1c4c2f3f95b78bc366"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e1cb9bf9ff95b194727697ba4789b26e"
                ],
                "markers": [
                    [
                        {
                            "id": [
                                "analyse_samplesheet.csv:md5,bf995bff1feac8b4fc2b4f46e7ae5b08"
                            ]
                        },
                        "test_markers.csv:md5,4b612101a8ba4a1c4c2f3f95b78bc366"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e1cb9bf9ff95b194727697ba4789b26e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-19T12:11:54.556332537"
    }
}